cmd/query/app/flags.go

```diff
@@ -36,20 +36,24 @@ import (
 )
 
 const (
+	queryHostPort           = "query.host-port"
 	queryPort               = "query.port"
 	queryBasePath           = "query.base-path"
 	queryStaticFiles        = "query.static-files"
 	queryUIConfig           = "query.ui-config"
 	queryTokenPropagation   = "query.bearer-token-propagation"
 	queryAdditionalHeaders  = "query.additional-headers"
 	queryMaxClockSkewAdjust = "query.max-clock-skew-adjustment"
+	queryPortWarning        = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
 )
 
 // QueryOptions holds configuration for query service
 type QueryOptions struct {
-	// Port is the port that the query service listens in on
+	// HostPort is the host:port address that the query service listens o n
+	HostPort string
+	// Port is the port that the query service listens in on ()
 	Port int
```

Comment:
I think this field should be removed, only the legacy CLI option should remain

Reply:
This field is necessary so the legacy CLI option is still functional. I will add deprecation notice to it.

Reply:
Why is it necessary? It is no longer assigned in InitFromViper

Reply:
I will look into it a bit more. So far the test fail with ``FAIL	github.com/jaegertracing/jaeger/cmd/query/app [build failed]``.

Reply:
Fixed it, waiting for Travis